containers/unit-tests/start

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+
+top_srcdir=$(realpath -m "$0"/../../..)
+image=cockpit/unit-tests:latest
+flags=''
+cmd=''
+
+while test $# -gt 0; do
+  case "$1" in
+    CC=*) ccarg="--env=$1";;
+    shell) flags=-it; cmd=/bin/bash;;
+    :*) image=cockpit/unit-tests"$1";;
+    *) image="$1";;
+  esac
+  shift
+done
+
+set -ex
+exec sudo docker run --shm-size=512M --volume "${top_srcdir}":/source:ro \
+       ${ccarg:+"$ccarg"} $flags -- "$image" $cmd
```

Comment:
What does this ccarg syntax do? should be fine to just do `${ccarg}`, if it's empty it just won't do anything?

Reply:
The problem is if someone specifies a C compiler with a space in the name — then it will expand to multiple arguments.  Pedantic, sure....

Reply:
Nice trick, thank you!